cdxj_indexer/main.py

```diff
@@ -360,22 +364,59 @@ class CDX09Indexer(CDXLegacyIndexer):
 
 # ============================================================================
 class SortingWriter:
-    def __init__(self, out):
+    MAX_SORT_BUFF_SIZE = 1024 * 1024 * 32
+
+    def __init__(self, out, max_sort_buff_size=None):
         self.out = out
         self.sortedlist = []
+        self.count = 0
+        self.max_sort_buff_size = max_sort_buff_size or self.MAX_SORT_BUFF_SIZE
+
+        self.tmp_files = []
 
     def write(self, line):
         self.sortedlist.append(line)
+        self.count += len(line)
+
+        if self.count > self.max_sort_buff_size:
+            self.tmp_files.append(self.write_to_temp())
+            self.sortedlist = []
+            self.count = 0
 
     def flush(self):
+        if not len(self.tmp_files):
+            self.sortedlist.sort()
+            self.write_to_file(self.sortedlist, self.out)
+            return
+
+        if len(self.sortedlist) > 0:
+            self.tmp_files.append(self.write_to_temp())
+            self.sortedlist = []
+            self.count = 0
+
+        open_files = [open(name, "rt") for name in self.tmp_files]
```

Comment:
I'm just double checking that it is ok to open as "t" here if we want the CDXJ file to be sorted in byte order?

Reply:
Yeah, as cdx are treated as text throughout, eg. https://github.com/webrecorder/cdxj-indexer/blob/main/cdxj_indexer/main.py#L295

But, perhaps we should have a set encoding, `encoding=utf-8` for open()?

Reply:
Based on this answer here: https://unix.stackexchange.com/a/564305 it seems like ensuring utf-8 output is compatible with byte-order sorting with LC_ALL=C that is generally recommended for cdx sorting.

Reply:
And even without setting it, at least on my Linux and OSX tests, python 3.8 `locale.getpreferredencoding()` returns 'UTF-8' so that's the default encoding.

Reply:
Great! Explicitly setting it to `utf-8` seems like the right thing to do, if it's easy?